src/plugins/piral-oidc/src/types.ts

```diff
@@ -215,3 +225,22 @@ export enum OidcErrorType {
 export interface PiralOidcError extends Error {
   type: Readonly<OidcErrorType>;
 }
+
+export interface SignInRedirectParams {
+  /**
+   * Values used to maintain state between the sign in request and the callback.
+   * These will be available on the result from the handleAuthentication function
+   * successfully authenticates from a callback state.
+   */
+  state?: PiralCustomRedirectState;
```

Comment:
Is this automatically stringified (i.e., JSON stringified and URL encoded)? Would a type `string` be more suitable (to enforce the user choosing the appropriate stringification)?

Reply:
From what I saw, oidc-client will actually store the state object you pass it as-is in session storage and pass a hash/id to the authorize endpoint. It will tie this id to the state object, so when you eventually return to the callback it will just grab the object off of local storage return it with the user.

Having said all of that we can enforce this is stringified for better future proofing if you want to go down that route

Reply:
Ah great - thanks for the explanation.

Other thought; should we put `any` here? People may use this for anything actually - not just storing some redirect object. It could be a string, boolean, whatever. Yes, most likely it will be an object, but we do not know that, right?

Reply:
Just so we're on the same page you're talking about changing the `state` properties on `SignInRedirectParams` and `AuthenticationResult` to be `any`, right? 

I also just tested this - For some reason I thought oidc-client wouldn't store "primitive" values as state, but it actually does allow this.

I do think having some typing for the state of the return result from the `handleAuthentication` call can be nice here. Maybe a generic on the function? If that is too complex or not obvious enough I'm perfectly okay with using `any` here.

Reply:
> Just so we're on the same page you're talking about changing the state properties on SignInRedirectParams and AuthenticationResult to be any, right?

Yes.

> I do think having some typing for the state of the return result from the handleAuthentication call can be nice here. Maybe a generic on the function?

It depends - I thought the same, but since there is no direct connection between calling this and having the state "returned" I'd say `any` is sufficiently good. But we could go for a generic function using `<T = any>`.

Reply:
I'll just go with any if you're okay with it!

Reply:
Yes - that's superb. Thanks! 🍻

Reply:
Now typed as any, should be good now 👍